pkg/render/node.go

```diff
@@ -486,16 +495,29 @@ func (c *nodeComponent) clusterAdminClusterRoleBinding() *rbacv1.ClusterRoleBind
 	return crb
 }
 
-// nodeDaemonset creates the node damonset.
+// nodeDaemonset creates the node daemonset.
 func (c *nodeComponent) nodeDaemonset(cniCfgMap *v1.ConfigMap) *apps.DaemonSet {
 	var terminationGracePeriod int64 = 0
+	var initContainers []v1.Container
 
 	annotations := make(map[string]string)
 	if len(c.birdTemplates) != 0 {
 		annotations[birdTemplateHashAnnotation] = AnnotationHash(c.birdTemplates)
 	}
 	annotations[typhaCAHashAnnotation] = AnnotationHash(c.typhaNodeTLS.CAConfigMap.Data)
-	annotations[nodeCertHashAnnotation] = AnnotationHash(c.typhaNodeTLS.NodeSecret.Data)
+	if c.cr.CertificateManagement == nil {
+		annotations[nodeCertHashAnnotation] = AnnotationHash(c.typhaNodeTLS.NodeSecret.Data)
```

Comment:
:thinking: If we are no longer using annotations to manage certificates ... Do we delete older annotations once you apply a CertificateManagement CR ?

Reply:
The operator already takes care of this, the node daemonset would be deployed and the previous annotations would be overwritten (e.g. if the new dameonset does not have an annotation it will not be there in the deployed daemonset).